api-get-object-attributes.go

```diff
@@ -0,0 +1,133 @@
+package minio
+
+import (
+	"context"
+	"encoding/xml"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/minio/minio-go/v7/pkg/encrypt"
+	"github.com/minio/minio-go/v7/pkg/s3utils"
+)
+
+// ObjectAttributesOptions is an API call that combines
+// HeadObject and ListParts.
+//
+// VersionID - The object version you want to attributes for
+// ServerSideEncryption - The server-side encryption algorithm used when storing this object in Minio
+type ObjectAttributesOptions struct {
+	VersionID            string
+	ServerSideEncryption encrypt.ServerSide
+}
+
+// ObjectAttributes ...
+type ObjectAttributes struct {
+	ObjectAttributesResponse
+	LastModified time.Time
+	VersionID    string
+}
+
+func (o *ObjectAttributes) parseResponse(resp *http.Response) (err error) {
+	mod, err := parseRFC7231Time(resp.Header.Get("Last-Modified"))
+	if err != nil {
+		return err
+	}
+	o.LastModified = mod
+	o.VersionID = resp.Header.Get(amzVersionID)
+
+	response := new(ObjectAttributesResponse)
+	if err := xml.NewDecoder(resp.Body).Decode(response); err != nil {
+		return err
+	}
+	o.ObjectAttributesResponse = *response
+
+	return
+}
+
+// ObjectAttributesResponse ...
+type ObjectAttributesResponse struct {
+	ETag         string `xml:",omitempty"`
+	StorageClass string
+	ObjectSize   int
+	Checksum     struct {
+		ChecksumCRC32  string `xml:",omitempty"`
+		ChecksumCRC32C string `xml:",omitempty"`
+		ChecksumSHA1   string `xml:",omitempty"`
+		ChecksumSHA256 string `xml:",omitempty"`
+	}
+	ObjectParts struct {
+		PartsCount int
+		Parts      []*ObjectAttributePart `xml:"Part"`
+	}
+}
+
+// ObjectAttributePart ...
+type ObjectAttributePart struct {
+	ChecksumCRC32  string `xml:",omitempty"`
+	ChecksumCRC32C string `xml:",omitempty"`
+	ChecksumSHA1   string `xml:",omitempty"`
+	ChecksumSHA256 string `xml:",omitempty"`
+	PartNumber     int
+	Size           int
+}
+
+// GetObjectAttributes ...
```

Comment:
Make this proper documentation. Eg

```suggestion
// GetObjectAttributes provides detailed object information.
```